packages/data-stores/src/auth/README.md

```diff
@@ -0,0 +1,83 @@
+# User store
+
+This store includes behavior for logging users in; including "passwordless" logins, 2-factor auth, etc.
+
+The store manages where the user is in login process. The login UI is responsible for displaying the appropriate components (e.g. password form, 2-factor code form) based on the state of the store, and the store is responsible for advancing through the flow depending on the authentication method the user is using.
+
+## Usage
+
+Register the auth store, passing in a valid client ID and secret:
+
+```js
+import { Auth } from '@automattic/data-stores';
+
+const AUTH_STORE = Auth.register( {
+	client_id: 'MY_CLIENT_ID', // ⚠️Replace with your app's ID
+	client_secret: 'MY_CLIENT_SECRET', // ⚠️Replace with your app's secret
+} );
+```
+
+### Login flow state
+
+The user can be in the following login states:
+
+- `ENTER_USERNAME_OR_EMAIL`
+- `ENTER_PASSWORD`
+- `ENTER_2FA_CODE`
+- `LOGIN_LINK_SENT`
+- `WAITING_FOR_2FA_APP`
+- `LOGGED_IN`
```

Comment:
hmm I've got a few questions about the states: 

Are all of the states definitely mutually exclusive? e.g. Can a user be in ENTER_2FA_CODE and LOGIN_LINK_SENT at the same time? 

I noticed that you can access the existing `log-in` page while already being `LOGGED_IN` in to an existing account and it gives you the option to e.g. "Continue as Eru Penkman" or log in as someone different.

Is the plan to have the states persisted between browser reloads? e.g. if a user were in the ENTER_2FA_CODE state and reloaded the app, what should happen? What if they're in the LOGGED_IN state?

Reply:
> Are all of the states definitely mutually exclusive? e.g. Can a user be in ENTER_2FA_CODE and LOGIN_LINK_SENT at the same time? 

Hmm these really are more UI states then they are ... states of the world (for want of a better word!) The `LOGIN_LINK_SENT` state is supposed mean "you should show the login link sent UI", not "in the past a login link has been sent".

Maybe part of the problem is what I called it: `loginFlowState`. I've been trying to think of something better 😅

> I noticed that you can access the existing `log-in` page while already being `LOGGED_IN` in to an existing account and it gives you the option to e.g. "Continue as Eru Penkman" or log in as someone different.

Nice spotting! Hadn't thought of that case.

What do you think of the auth store ignoring existing sessions? The login components could use the user store to show the the "Continue as Alex" button?

> Is the plan to have the states persisted between browser reloads?

The current `/log-in` UI throws all the state away when the browser reloads. My first reaction is just to copy that behaviour. For gutenboarding it'll be in the modal, and the modal will also close when the page refreshes.

But if we wanted to change that I think it's just a flag you can pass to `registerStore()`

Reply:
ahh nice both of those answers make sense, yea so it's more the state of the login page then the state of the world.. yea I do think `loginFlowState` is a good name.

And yep that sounds like it would work to have the auth store ignore any existing session and managing the already logged in case from the component level.

👍

Reply:
❤️ the idea of using states. I wonder if we could use something similar for https://github.com/Automattic/wp-calypso/issues/39268

For example, if a user is in the `NEW_SITE_CREATED` state we know not to allow site creation.

Or something.